CRM/Report/Form/Contact/Relationship.php

```diff
@@ -629,6 +629,39 @@ public function alterDisplay(&$rows) {
 
     foreach ($rows as $rowNum => $row) {
 
+      // Handle ID to label conversion for contact fields
+      $entryFound = $this->alterDisplayContactFields($row, $rows, $rowNum, 'contact/relationship', 'View Relationships') ? TRUE : $entryFound;
+
+      // Handle contact subtype A
+      if (array_key_exists('civicrm_contact_contact_sub_type_a', $row)) {
```

Comment:
This is probably OK but it repeats the same thing twice. It's not a show stopper - but ideally you would refactor into a separate function (possibly shared by the parent class sub type handling) & call that rather than repeat it.

I'm happy to leave that out of scope this round - but perhaps you could comment it as a @todo